src/parser/parser.cc

```diff
@@ -7,12 +7,15 @@ using lexer::TokenType;
 Parser::Parser(lexer::Lexer* lexer)
   : lexer_(lexer) {}
 
-ParseError Parser::parse() {
-  Token t = lexer_->get_next_token();
+MaybeError<ParseError> Parser::parse() {
+  auto res = lexer_->get_next_token();
+  if (!res.ok())
+    return {res.error_or_die()};
```

Comment:
This kind of construct (get result and propagate in case of error) will certainly be common, what about creating a macro for it? This would avoid some potential stupid errors (like forgetting the ! in the condition).